### PR TITLE
[chore] Update otel core libraries

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Handle breaking change in confignet

Locally I had to switch `configcompression` back to `0.96.0` instead of `1.3.0` because it hadn't been tagged with that yet. When running `make update-otel` the generated version I was getting was `1.3.1-*` because it expected a `1.3.0` version of the module to have already been released.

### CODE REVIEW

**Comment by crobert-1** [comment]
**Overview:**
I found the issue with the failing `TestPostgresIntegrationLogsTrackingWithStorage`. 

The test is creating a receiver twice, using the same storage extension. The first receiver reads all the logs in the file, and saves state with the storage extension. The test then creates a new receiver and expects its "spot" in the file to be saved, and to not read any more logs.

The issue is [this line](https://github.com/open-telemetry/opentelemetry-collector/blob/3b5aee0c7a168f7ba57a87e864b4f65d794eef8b/receiver/receivertest/nop_receiver.go#L22), introduced in https://github.com/open-telemetry/opentelemetry-collector/pull/9637. When [starting a receiver with a storage extension](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d82d6f5adf83c678a7a11ce3f151a3530eec6708/receiver/sqlqueryreceiver/logs_receiver.go#L85) the receiver's newly-unique ID is eventually [used to determine the file path for the extension's state](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d82d6f5adf83c678a7a11ce3f151a3530eec6708/extension/storage/storagetest/client.go#L53).

Now that `nop` receivers are given a unique ID for each instance, when creating a second receiver, it has a different unique ID, resulting in looking at a different file for its state. With a new file it's no longer able to load the tracking state and starts over. That's why we're seeing the test failure there.

**Summary:**
This is fallout from https://github.com/open-telemetry/opentelemetry-collector/pull/9637. The fix is essentially the same thing already done [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31659), I'll just adapt this solution to this test.

**Comment by TylerHelmuth** [comment]
Needs https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31792 to unblock CI

**Comment by TylerHelmuth** [comment]
@open-telemetry/collector-contrib-approvers this is finally passing ci and ready for review.

**Review by fatsheep9146 — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

**Pull request merged**
